### PR TITLE
Adding NodeKeeper Controller block in design.md

### PR DESCRIPTION
**Pull request by anjoshi24**
Signed-off-by: Aniket Joshi <anjoshi@redhat.com>

### What type of PR is this?
Documentation

### What this PR does / why we need it?
Introduces the NodeKeeper Controller into the UpgradeOperator Documentation.
Few introductory details of the NodeKeeper Controller have been added in the Design Doc.

### Which Jira/Github issue(s) this PR fixes?
[OSD-3472](https://issues.redhat.com/browse/OSD-3472)

### Pre-checks (if applicable):
- [ ] Tested latest changes against a cluster
- [ ] Ran `make generate` command locally to validate code changes
- [ ] Included documentation changes with PR

### CODE REVIEW

File: docs/design.md [hunk 1]
```diff
@@ -109,11 +109,19 @@ A fully-populated example of an `UpgradeConfig` status is included below:
...
 
 For more information, see the dedicated section on this topic: [UpgradeConfig Managers](configmanager.md)
 
+## Controllers
+
+The `managed upgrade operator` provided upgrade process revolves around multiple Controllers. Alongside the above mentioned `UpgradeConfig` controller, the `NodeKeeper` controller works simultaneously in an upgrade process towards the state of nodes in the cluster.
+
+The `NodeKeeper` controller keeps a track of state of all the nodes during an upgrade, making sure of any activities around the nodes and acts upon any action required towards maintaining the required state of any node.
+
+If a node is having any troubles while an ongoing upgrade, the `NodeKeeper` controller will perform necessary action towards the state of the node and upgrade continuation.
```
Comment (mrbarge) [inline docs/design.md:123]:
```suggestion
If an upgrading worker node is experiencing difficulty draining due to conditions such as [Pod Disruption Budgets](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/#pod-disruption-budgets) or stuck finalizers), the `NodeKeeper` controller will perform remediation strategies to ensure the node's eventual drain and subsequent upgrade continuation.

The `NodeKeeper` controller will flag through metrics any worker node that continue to unsuccessfully drain in spite of the remediation strategies. 
```

File: docs/design.md [hunk 2]
```diff
@@ -109,11 +109,19 @@ A fully-populated example of an `UpgradeConfig` status is included below:
...
+The `managed-upgrade-operator` provides a configurable mechanism for retrieving and storing an `UpgradeConfig`
 Custom Resource that can be reconciled against by the `UpgradeConfig` controller.
 
 For more information, see the dedicated section on this topic: [UpgradeConfig Managers](configmanager.md)
 
+## Controllers
+
+The `managed upgrade operator` provided upgrade process revolves around multiple Controllers. Alongside the above mentioned `UpgradeConfig` controller, the `NodeKeeper` controller works simultaneously in an upgrade process towards the state of nodes in the cluster.
+
+The `NodeKeeper` controller keeps a track of state of all the nodes during an upgrade, making sure of any activities around the nodes and acts upon any action required towards maintaining the required state of any node.
```
Comment (mrbarge) [inline docs/design.md:121]:
```suggestion
The `NodeKeeper` controller keeps a track of the upgrading worker nodes during an upgrade and seeks to ensure their timely and eventual upgrade.
```

Comment (mrbarge) [reply docs/design.md:121]: Just a couple of suggestions here to help clarify that it's worker nodes specifically.

File: docs/design.md (hunk 1 above)
Comment (anjoshi24) [reply docs/design.md]: Thanks for the suggestion, I have made changes according to the suggestion.

File: docs/design.md (hunk 2 above)
Comment (anjoshi24) [reply docs/design.md]: Thanks for the suggestion, I have made changes according to the suggestion.

**Pull request merged**